node/src/types/deploy.rs

```diff
@@ -474,16 +477,33 @@ impl Deploy {
         &self.approvals
     }
 
-    /// Returns the header() wrapped the correct DeployType given the `session`.
+    /// Returns the DeployType wrapped the correct DeployType.
     pub fn deploy_type(&self) -> DeployType {
-        match self.session() {
-            ExecutableDeployItem::Transfer { .. } => DeployType::Transfer(self.header().clone()),
-            ExecutableDeployItem::ModuleBytes { .. }
-            | ExecutableDeployItem::StoredContractByHash { .. }
-            | ExecutableDeployItem::StoredContractByName { .. }
-            | ExecutableDeployItem::StoredVersionedContractByHash { .. }
-            | ExecutableDeployItem::StoredVersionedContractByName { .. } => {
-                DeployType::Wasm(self.header().clone())
+        let header = self.header().clone();
+        let size = self.serialized_length();
+        if self.session().is_transfer() {
+            // TODO: non-zero value
+            let payment_amount = Motes::zero();
+            DeployType::Transfer {
+                header,
+                payment_amount,
+                size,
+            }
+        } else {
+            let payment_item = self.payment().clone();
+            let payment_amount = {
+                // Some assumptions here:
+                // Since we are a Transfer variant, we expect: args to exist, contain "amount", and
```

Comment:
You mean we are _not_ a transfer?

Reply:
Nice catch, I left the comment out of date, thanks.